### ISSUE TITLE
improvement proposal: dynamic motor power allocation

### ISSUE THREAD

**Issue by dylansison** [LeoDog896/Landstown-Robotics-Challenge#10]
**The ability to dynamically allocate power to ROV thrusters** will allow us to maximize the use of all 12 volts and 25 amps of power through the allocation of a motor amperage total, based on the power envelope remaining after all other electrical subsystems' requirements are accounted for.

Total motor power = motor1 + motor2 + motor3 + motor4 + motor5 + motor6 

At the moment, each motor is limited to about 50% power, to draw 3.3a/motor max. This provides 1.28kg/f thrust/motor. In the forward direction, 2.56kg/f is applied when maximum power is requested. This solution would allow each motor to pull half of the power available to motors.
let motor power allocation = 20amps
2 forward motors = 10amps available/motor
per https://bluerobotics.com/store/thrusters/t100-t200-thrusters/t200-thruster-r2-rp/:
at 10amps, 1 t200 thruster provides ~2.62kg/f thrust. 
forward motor thrust = 5.24kg/f **Just over 2x the thrust**

As with all solutions, this comes with downsides. Overall motor thrust may diminish as more motors are activated. Additionally, complex software control of thrusters provides an additional point of failure. This may be addressed through an easily activatable fallback/legacy mode.

TL;DR dynamic power allocation = use closer to all 25 amps

EDIT 1: This is more important now that we're planning on running 6 motors; if you divided power in a similar way, our value would be around 1/3 instead of 1/2 of the true max

(sorry Tristan, dunno how this fits under conventional commits)
